### ISSUE TITLE
Create accessible multi-select autocomplete in Storybook

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#15421]
The current "multiple selection" autocomplete for tags in the V2 editor has very poor accessibility - it creates a focus trap meaning you can never reach the editor body by keyboard, and lacks any of the expected semantics for a combobox control.

This issue lays the foundation for the UI improvements to the tag autosuggest by creating an accessible multi-select autocomplete component that we can then build upon.

## Not included in this issue

This issue lays groundwork only. What the 'suggestions' look like will be addressed in a later issue. For the moment, it is sufficient if suggestions appear, are selectable, editable and removable as per below.

## Technical approach

Unfortunately the Reach UI Combobox library that we use for `MentionAutocomplete` functionality does not support multiple selection. However, as we intend to remove already-added-tags from the suggestion list, we might be able to still use a "single selection" combobox like this. 

Even if not,  there has been some [great exploration into accessible, maintainable patterns for this kind of widget by Sarah Higley](https://www.24a11y.com/2019/select-your-poison-part-2/), including [some demo code](https://codepen.io/smhigley/pen/GRgjRVN) very close to what we are trying to achieve.

I think we could "roll our own" here if needed, making sure we conform with the [combobox authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox). This may well pave the way for us later removing the dependency on Reach UI Combobox for MentionAutocomplete, but that is not in the scope of this current project.

As we build this out we should do it with a view to being able to re-use the core autosuggest functionality in other places.

## Expected features

[Please see the full detail in the Figma file](https://www.figma.com/file/RzV8pYcgujohbdTjYOKqld/Tags-autocomplete?node-id=0%3A1)

- User should be able to select an item from multiple options in a list
- On selection, the item should appear with an 'X' icon next to it
- Clicking the 'X' should remove the selection
- Clicking on the selection's text, rather than the 'X' should place it back in 'edit mode'
- On selecting an item, text should be cleared and a new selection should be possible by continuing to type
- Typing a comma or space, should have the effect of selecting the currently entered text
- Pressing delete should place the last selection (if any) into 'edit mode'

Keyboard interaction patterns etc should match up with the [combobox authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox)
